src/libraries/System.Private.CoreLib/src/System/IO/Path.Windows.cs

```diff
@@ -27,6 +27,39 @@ public static partial class Path
             (char)31
         };
 
+        // Checks if the given path is available for use.
+        public static bool Exists([NotNullWhen(true)] string? path)
+        {
+            try
+            {
+                if (string.IsNullOrEmpty(path))
+                {
+                    return false;
+                }
+
+                string fullPath = GetFullPath(path);
+                Interop.Kernel32.WIN32_FILE_ATTRIBUTE_DATA data = default;
+                int errorCode = FileSystem.FillAttributeInfo(fullPath, ref data, returnErrorOnNotFound: true);
```

Comment:
the `FileSystem.FillAttributeInfo` method returns an error code, it does not throw:

https://github.com/dotnet/runtime/blob/11bf8a782d27dd0f614dd0da03b27ab025933320/src/libraries/Common/src/System/IO/FileSystem.Attributes.Windows.cs#L112

It's more of a `C` than a `C#` method. 

Since `GetFullPath` can throw and we want to be consistent with `File.Exists`:

>  This method also returns false if path is null, an invalid path, or a zero-length string.

We still need the catch block. Since you need to add triple slash comments to document the new API and we don't want to duplicate them for Windows and Unix implementaiton, you can move the common logic to `Path.cs` file and keep the OS-specific in `Path.Winows|Unix.cs`.

Sth like this:

```cs
// Path.cs

/// <summary>
/// 
/// </summary>
/// <remarks>
/// 
/// </remarks>
/// <param name="path"></param>
/// <returns></returns>
public static bool Exists([NotNullWhen(true)] string? path)
{
    if (string.IsNullOrEmpty(path))
    {
        return false;
    }
    
    string? fullPath;
    try
    {
        fullPath = GetFullPath(path
    }
    catch (ArgumentException)
    {
        return false;
    }
    catch (IOException)
    {
        return false;
    }
    
    return ExistsCore(fullPath);
}

// Path.Windows.cs
private static bool ExistsCore(string fullPath)
{
    Interop.Kernel32.WIN32_FILE_ATTRIBUTE_DATA data = default;
    int errorCode = FileSystem.FillAttributeInfo(fullPath, ref data, returnErrorOnNotFound: true);
   // the rest of the logic 
}
```

Reply:
Fixed